RELEASE/scripts/excavator/projects/x_genie.ash

```diff
@@ -0,0 +1,94 @@
+/**
+ * @author gausie
+ * Determine which monsters and effects can be fought/acquired with the Genie.
+ */
+
+import <excavator/x_utils.ash>
+
+void spade_genie( string page )
+{
+    // Check page validity
+    if ( !page.contains_text( "<b>Results:</b>" ) )
+    {
+        return;
+    }
+
+    // Find our wish or bail out
+    matcher m = "You announce, \"I wish (.*?)\"<".create_matcher( page );
+
+    if ( !m.find() )
+    {
+        return;
+    }
+
+    string wish = m.group( 1 );
+
+    // Determine if this was a success
+    boolean success = !page.contains_text( "<br />You could try, " );
+
+    string [string] data = {
+        "success": success,
+    };
+
+    boolean wishable;
+
+    matcher fight_matcher = "(?:to (?:fight|be fighting) (?:a )?)(.*)".create_matcher( wish );
+
+    // Determine wishability for fight and prepare data map
+	if ( fight_matcher.find() )
+    {
+        monster mon = fight_matcher.group( 1 ).to_monster();
+
+        data["type"] = "monster";
+        data["value"] = mon.to_string();
+
+        wishable = !mon.boss && mon.copyable;
+    }
+    // Determine wishability for effect and prepare data map
+    else if ( wish.starts_with( "to be " ) || wish.starts_with( "i was " ) )
```

Comment:
This will catch "i was rich" and others.

Reply:
do you think it would be alright to just bail if `eff == $effect[none]`? I mean it should do that anyway